From a28bfef4e846a7ea2482ac5d57185bc754ba6de3 Mon Sep 17 00:00:00 2001
From: EdenEast <edenofest@gmail.com>
Date: Fri, 13 Oct 2023 16:39:15 -0400
Subject: [PATCH] feat!: Use `XDG_CONFIG_HOME` if defined on macos

dirs_next::config_dir returns this path on macos:

`/Users/<user>/Library/Application Support`

If the user has defined `XDG_CONFIG_HOME` on macos then they expect
nushell to act like other *nix systems. This could be defined in a plist
file under `$HOME/Library/LaunchAgents` or from the root shell if
launched from another shell.
---
 crates/nu-path/src/helpers.rs | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/crates/nu-path/src/helpers.rs b/crates/nu-path/src/helpers.rs
index febe8a2a7..97edf1479 100644
--- a/crates/nu-path/src/helpers.rs
+++ b/crates/nu-path/src/helpers.rs
@@ -6,6 +6,19 @@ pub fn home_dir() -> Option<PathBuf> {
     dirs_next::home_dir()
 }
 
+#[cfg(target_os = "macos")]
+pub fn config_dir() -> Option<PathBuf> {
+    // dirs_next::config_dir returns the path `/Users/<user>/Library/Application Support` on macos.
+    // If the user has defined `XDG_CONFIG_HOME` on macos then they expect nushell to act like
+    // other *nix systems. This could be defined in a plist file under `$HOME/Library/LaunchAgents`
+    // or from the root shell if launched from another shell.
+    std::env::var("XDG_CONFIG_HOME")
+        .map(PathBuf::from)
+        .ok()
+        .or(dirs_next::config_dir())
+}
+
+#[cfg(not(target_os = "macos"))]
 pub fn config_dir() -> Option<PathBuf> {
     dirs_next::config_dir()
 }
-- 
2.42.0

